portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -116,28 +147,77 @@ function insertSearch() {
 
   const searchInput = document.createElement('input');
   searchInput.className = 'form-control form-inline';
+  searchInput.id = 'searchQuery';
   searchInput.type = 'text';
   searchInput.placeholder = 'Search';
 
   const searchI = document.createElement('i');
   searchI.className = 'fas fa-search text-white ml-3 mr-auto';
 
+  const searchResults = document.createElement('ul');
+  searchResults.id = 'searchResults';
+
   searchDiv.appendChild(searchInput);
   searchDiv.appendChild(searchI);
   searchBar.appendChild(searchDiv);
+  searchBar.append(searchResults);
 
-  document.getElementById('mainNav').appendChild(searchBar);
+  return searchBar;
 }
 
-async function fetchSubmittedLocations() {
-  const response = await fetch('/submitted-locations');
-  const userComments = await response.json();
-  const commentData = [];
+/**
+ * searches each child Node of the page in the docElements and retains the
+   elements that contain the wanted word.
+ */
+function searchPages(docElements, wantedWords) {
+  const numElements = docElements.length;
+  const resultElements = [];
+
+  if (wantedWords.length <= 0) {
+    return resultElements;
+  }
+
+  for (let i = 0; i < numElements; i++) {
```

Comment:
nit: can this be a forEach now? Also, does it need to be called frontElement now that you're only searching forward?

Reply:
Yeah sure, I changed it.